src/views/login/components/gmail-login/index.js

```diff
@@ -0,0 +1,62 @@
+import React, { Component, PropTypes } from 'react'
+import { connect } from 'react-redux'
+import { bindActionCreators } from 'redux'
+import SocialButton from '../social-button'
+import { authenticateWithGoogle } from '../../../../modules/auth/actions'
+
+import gmailLogo from '../../../../../assets/img/gmail-icon.png'
+
+import styles from './styles.css'
+
+const mapStateToProps = (state) => {
+  const { token } = state
+  return { token }
+}
+
+const mapDispatchToProps = (dispatch) => (
+  bindActionCreators({
+    authenticateWithGoogle,
+  }, dispatch)
+)
+
+@connect(mapStateToProps, mapDispatchToProps)
+class GmailLogin extends Component {
+  constructor(props) {
+    super(props)
+    this.clickHandler = this.clickHandler.bind(this)
```

Comment:
This is unnecessary if you write the class function as arrow-function.. see next comment